pkg/server/errors/errors.go

```diff
@@ -26,6 +26,7 @@ var (
 	MismatchObjectType                     = status.Error(codes.Code(openfgav1.ErrorCode_query_string_type_continuation_token_mismatch), "The type in the querystring and the continuation token don't match")
 	RequestCancelled                       = status.Error(codes.Code(openfgav1.InternalErrorCode_cancelled), "Request Cancelled")
 	RequestDeadlineExceeded                = status.Error(codes.Code(openfgav1.InternalErrorCode_deadline_exceeded), "Request Deadline Exceeded")
+	ThrottledTimeout                       = status.Error(codes.Code(openfgav1.UnprocessableContentErrorCode_throttled_timeout_error), "Request Deadline Exceeded while throttled")
```

Comment:
This message is inconsistent with our docs. I suggest the following rename and update the API doc to align with it:

> request deadline exceeded due to throttling

<img width="651" alt="Screenshot 2024-04-25 at 1 47 08 PM" src="https://github.com/openfga/openfga/assets/2899204/bd255c5a-5607-47b9-8435-12a434968aa9">